bitcoin/src/string.rs

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: CC0-1.0
+
+//! Bitcoin string parsing utilities.
+//!
+//! This module provides utility types and traits
+//! to support handling and parsing strings within `rust-bitcoin`.
+
+use crate::prelude::String;
+
+/// Hex parsing error
+#[derive(Debug, Eq, PartialEq, Clone)]
+#[non_exhaustive]
+pub enum FromHexError<E> {
+    /// The input was not a valid hex string, contains the error that occurred while parsing.
+    ParseHex(E),
+    /// The input is missing `0x` prefix, contains the invalid input.
+    MissingPrefix(String),
+}
+
+/// Trait that allows types to be initialized from hex strings
+pub trait FromHexStr: Sized {
+    /// An error occurred while parsing the hex string.
+    type Error;
+
+    /// Parses provided string as hex requiring 0x prefix.
+    ///
+    /// This is intended for user-supplied inputs or already-existing protocols in which 0x prefix is used.
+    fn from_hex_str<S: AsRef<str> + Into<String>>(s: S) -> Result<Self, FromHexError<Self::Error>> {
+        if !s.as_ref().starts_with("0x") {
+            Err(FromHexError::MissingPrefix(s.into()))
+        } else {
+            Self::from_hex_str_no_prefix(s.as_ref().trim_start_matches("0x"))
+                .map_err(FromHexError::ParseHex)
```

Comment:
This could be
```rust
            Ok(Self::from_hex_str_no_prefix(s.as_ref().trim_start_matches("0x"))?)
```
If we had
```rust

impl<E> From<E> for FromHexError<E> {
    fn from(e: E) -> Self {
        FromHexError::ParseHex(e)
    }
}
```

Reply:
For sure, this'll clean it up nicely. Added